app/controllers/application_controller.rb

```diff
@@ -1,10 +1,24 @@
 class ApplicationController < ActionController::Base
+  include Pundit
   include SameOriginSessionsConcern
 
   before_action :set_environment
 
+  rescue_from Pundit::NotAuthorizedError, with: :not_authorized
+
   private
 
+  def not_authorized
+    return render(json: {}, status: :unauthorized) if request.xhr?
```

Comment:
I'll make it `request.format == :json` for now.

Where my thoughts are going is to more cleanly separate out base `ApplicationController` for the HTML parts of the app and `ApiController` for the API parts and separately handle the authorization errors rather than having this one method cover both cases.

Reply:
That sounds like a good idea. 👍